server/controllers/todoController.js

```diff
@@ -0,0 +1,151 @@
+const { Todo } = require('../models')
+
+class Controller {
+    static showAll(req, res) {
+        Todo.findAll({
+            where: {UserId: req.user.id},
+            order: [['createdAt', 'ASC']]
+        })
+        .then(data => {
+            res.status(200).json(data);
+        })
+        .catch(err => {
+            res.status(500).json({message: "internal server error"});
+        })
+    }
+
+    static add(req, res) {
+        let input = {
+            title: req.body.title,
+            description: req.body.description,
+            status: req.body.status,
+            due_date: req.body.due_date,
+            UserId: req.user.id
+        }
+        Todo.create(input)
+        .then(data => {
+            let {id, title, description, status, due_date, UserId} = data
+            const options = { 
+                month: '2-digit', 
+                day: '2-digit',
+                year: 'numeric', 
+            };
+            let output = {
+                id,
+                title,
+                description,
+                status,
+                due_date: due_date.toLocaleString("en-ID", options),
+                UserId
+            }
+            res.status(201).json(output)
+        })
+        .catch(err => {
+            let message = []
+            err.errors.forEach(el => {
+                message.push(el.message)
+            });
+            if (message.length) {
+                return res.status(400).json({message})
+            }
+            res.status(500).json({message: "internal server error"})
+        })
+    }
+
+    static showOne(req, res) {
+        let id = req.params.id
+        Todo.findOne({
+            where: {
+                id,
+                UserId: req.user.id
+            }})
+        .then(data => {
+            if (data) {
+                res.status(200).json(data)
+            } else {
```

Comment:
else nya berarti sudah tidak perlu ya, karna not foundnya sudah di handle di middleware authorization